dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/event/EnrollmentController.java

```diff
@@ -326,6 +338,27 @@ public void deleteEnrollment( @PathVariable String id, HttpServletRequest reques
         webMessageService.send( WebMessageUtils.importSummary( importSummary ), response, request );
     }
 
+    // -------------------------------------------------------------------------
+    // QUEUED IMPORT
+    // -------------------------------------------------------------------------
+
+    @RequestMapping( value = "/queue", method = RequestMethod.POST, consumes = "application/json" )
+    public void postQueuedJsonEvents( @RequestParam( defaultValue = "CREATE_AND_UPDATE" ) ImportStrategy strategy,
+        HttpServletResponse response, HttpServletRequest request, ImportOptions importOptions ) throws Exception
+    {
+        if ( !trackerKafkaManager.isEnabled() )
+        {
+            throw new WebMessageException( WebMessageUtils.badRequest( "Kafka integration is not enabled." ) );
+        }
+
+        importOptions.setImportStrategy( strategy );
+        importOptions.setIdSchemes( getIdSchemesFromParameters( importOptions.getIdSchemes(), contextService.getParameterValuesMap() ) );
+
+        List<Enrollment> enrollments = enrollmentService.getEnrollmentsJson( StreamUtils.wrapAndCheckCompressionFormat( request.getInputStream() ) );
+        JobConfiguration job = trackerKafkaManager.dispatchEnrollments( currentUserService.getCurrentUser(), importOptions, enrollments );
+        webMessageService.send( jobConfigurationReport( job ), response, request );
```

Comment:
We now have a WebMessageUtils.jobConfigurationReport method for reporting jobConfigurationReport!